### PR TITLE
Add Zitadel documentation for local development

### LINKED ISSUE #614: Add Zitadel documentation for local development
The following must be added to `xpanse-iam` repo under directory - https://github.com/eclipse-xpanse/xpanse-iam/tree/main/zitadel

- [ ] new folder `local`
- [ ] Add steps on how to start the docker containers for Zitadel
- [ ] How to run the configuration scripts.
- [ ] How to use this xpanse and API

Please note - this user story is not only about documentation. We must test the process and then document all steps.

### PR DESCRIPTION
**Pull request by baixinsui**
fixed https://github.com/eclipse-xpanse/xpanse/issues/614

### CODE REVIEW

File: zitadel/local/README.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+
+Open the project of [Xpanse-ui](https://github.com/eclipse-xpanse/xpanse-ui.git) with the IDEA for local development.
+Copy the config file '.env.zitadel-testbed' and save as a new env file named '.env.zitadel-local' under the root path
+of the project `xpanse-ui`, then update the context of config item as below:
+
+* The value of item 'REACT_APP_ZITADEL_AUTHORITY_NAME' copied from item 'authorization-server-endpoint' in Outputs;
+* The value of item 'REACT_APP_ZITADEL_CLIENT_ID' copied from item 'authorization-xpanse-ui-client-id' in Outputs;
+![copy-env-file-update-item](compose/statics/copy-env-file-update-item.png)
+
+Edit the config file 'package.json', copy the scripts 'start-with-zitadel-testbed' and 'build-with-zitadel-testbed', and
```
Comment (swaroopar) [inline zitadel/local/README.md]: we must remove this step. No temporary changes to package.json is preferred. 

by default the `npm start` command must use the values in .env file. So it is enough to update this file to point to port 8081

https://github.com/swaroopar/xpanse-ui/blob/5ecb08b8486caa2068f97770b78d80dad98fed1b/.env#L2

File: zitadel/local/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+![authorize-on-swagger-ui.png](compose/statics/authorize-on-swagger-ui.png)
+
+### Config and run the UI client of Xpanse.
+
+Open the project of [Xpanse-ui](https://github.com/eclipse-xpanse/xpanse-ui.git) with the IDEA for local development.
+Copy the config file '.env.zitadel-testbed' and save as a new env file named '.env.zitadel-local' under the root path
+of the project `xpanse-ui`, then update the context of config item as below:
+
+* The value of item 'REACT_APP_ZITADEL_AUTHORITY_NAME' copied from item 'authorization-server-endpoint' in Outputs;
+* The value of item 'REACT_APP_ZITADEL_CLIENT_ID' copied from item 'authorization-xpanse-ui-client-id' in Outputs;
```
Comment (swaroopar) [inline zitadel/local/README.md]: REACT_APP_ZITADEL_CLIENT_ID -> we can mention how to add this as environment variable and start the app.

REACT_APP_ZITADEL_AUTHORITY_NAME -> we do not need this. just hve to update .env file.

File: zitadel/local/README.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+2. Select the project `eclipse-xpanse`and then click on the button `Continue`;
+![img.png](compose/statics/grant-select-project.png)
+3. Select all roles to the grant and then clock on the button `Save`;
+![img.png](compose/statics/grant-select-all-roles.png)
+After completing the grant above, user has roles as below:
+![img.png](compose/statics/view-user-has-all-roles.png)
+
+## Run Applications of Xpanse with Local Service of Zitadel
+
+### Config and run the API service of Xpanse.
```
Comment (swaroopar) [inline zitadel/local/README.md:129]: same here. I would not recommend temporarily updating any files that are checked-in into repo.
Even here, we must only show how to provide these values as environment variables and start the server. 

1. for fixed values such as url, we can permanently checkin the zitadel-local properties file. 
2. start server with zitadel  and zitadel-local profile and env variables for Zitadel properties. 

I think we can also remove 'zitadel-testbed' from default profiles. Then the developer must explicitly select what he wants to do.

**Review by swaroopar — CHANGES_REQUESTED** [review]
great work. Just changes required for the way we use local configurations.

File: zitadel/local/README.md (hunk 1 above)
Comment (baixinsui) [reply zitadel/local/README.md]: Updated it by https://github.com/eclipse-xpanse/xpanse-ui/pull/170

File: zitadel/local/README.md (hunk 2 above)
Comment (baixinsui) [reply zitadel/local/README.md]: Ok.

File: zitadel/local/README.md (hunk 3 above)
Comment (baixinsui) [reply zitadel/local/README.md:129]: Updated it by https://github.com/eclipse-xpanse/xpanse/pull/786.  & https://github.com/eclipse-xpanse/xpanse-website/pull/32

**Review by swaroopar — APPROVED** [review]
LGTM

**Pull request merged**